src/qm.rs

```diff
@@ -2976,16 +1681,19 @@ static qm_tbl_32x32_t: [[[u8; (32 * (32 + 1)) / 2]; 2]; 15] = [
 
 const fn subsampled<const N: usize, const M: usize>(
     src: &[u8; N],
-    sz: usize,
-    step: usize,
+    h: usize,
+    hstep: usize,
+    vstep: usize,
 ) -> [u8; M] {
-    assert!((sz * step) * (sz * step) == N);
-    assert!(sz * sz == M);
+    assert!(N == h * 32);
+    assert!(M * hstep * vstep == h * 32);
     let mut dst = [0; M];
+    let mut i = 0;
 
-    const_for!(y in 0..sz => {
-        const_for!(x in 0..sz => {
-            dst[y * sz + x] = src[y * sz * step * step + x * step];
+    const_for!(y in 0..h, step_by vstep => {
+        const_for!(x in 0..32, step_by hstep => {
+            dst[i] = src[(y + (vstep - 1) / 2) * 32 + x + (hstep - 1) / 2];
```

Comment:
```suggestion
            dst[i] = src[y * 32 + x];
```

Why can't we do it simpler like C is?

Reply:
The C code adds an offset to the `src` pointer before passing it to this function. I avoided doing so to simplify the array size check in this function. I could do `src = &src[(vstep - 1) / 2) * 32 + (hstep - 1) / 2..]` before entering this loop or add an offset parameter to this function. Any preference?

Reply:
Ahh.  I didn't think slicing works in a `const fn`, but I think storing a separate `offset` that we add to the index each time would be easier to read.